RequestError [HttpError]: Cookies must be enabled to use GitHub.

## Current behavior

<!-- Describe how the issue manifests. -->

I set-up a semantic-release workflow via GitHub actions for the repository https://github.com/kleinfreund/vue-accessible-color-picker. On push, the workflow was run and the first version of my package was published (yay!). However, the next step failed (full error below):

```
[semantic-release] › ✖  Failed step "publish" of plugin "@semantic-release/github"
[semantic-release] › ✖  An error occurred while running semantic-release: RequestError [HttpError]: Cookies must be enabled to use GitHub.
```

I’m not sure what to make of this error. I don’t recall reading anything in the getting started documents and articles I followed about some cookie settings I needed to be aware of.

Note that I have both `NPM_TOKEN` and `GH_TOKEN` configured via the secrets settings of the repository in question. The `GH_TOKEN` is a public access token which was created with the `public_repo` permission.

What am I missing from my setup and how can I recover from this botched release?

<details>
<summary>full error</summary>

[semantic-release] › ✖  Failed step "publish" of plugin "@semantic-release/github"
[semantic-release] › ✖  An error occurred while running semantic-release: RequestError [HttpError]: Cookies must be enabled to use GitHub.
    at /home/runner/work/vue-accessible-color-picker/vue-accessible-color-picker/node_modules/@octokit/request/dist-node/index.js:66:23
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  status: 403,
  headers: {
    'cache-control': 'no-cache',
    connection: 'close',
    'content-encoding': 'gzip',
    'content-security-policy': "default-src 'none'; base-uri 'self'; block-all-mixed-content; connect-src 'self' uploads.github.com www.githubstatus.com collector.githubapp.com api.github.com www.google-analytics.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com cdn.optimizely.com logx.optimizely.com/v1/events wss://live.github.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com; frame-ancestors 'none'; frame-src render.githubusercontent.com; img-src 'self' data: github.githubassets.com identicons.github.com collector.githubapp.com github-cloud.s3.amazonaws.com *.githubusercontent.com; manifest-src 'self'; media-src 'none'; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com",
    'content-type': 'text/plain; charset=utf-8',
    date: 'Thu, 14 May 2020 17:40:28 GMT',
    'expect-ct': 'max-age=2592000, report-uri="https://api.github.com/_private/browser/errors"',
    'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
    server: 'GitHub.com',
    'set-cookie': '_gh_sess=[REDACTED]; path=/; secure; HttpOnly',
    status: '403 Forbidden',
    'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
    'transfer-encoding': 'chunked',
    vary: 'X-PJAX, Accept-Encoding, Accept, X-Requested-With',
    'x-content-type-options': 'nosniff',
    'x-frame-options': 'deny',
    'x-github-request-id': '0400:0851:57C564:924595:5EBD828C',
    'x-xss-protection': '1; mode=block'
  },
  request: {
    method: 'POST',
    url: 'https://github.com/repos/kleinfreund/vue-accessible-color-picker/releases',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-rest.js/17.9.0 octokit-core.js/2.5.0 Node.js/12.16.3 (Linux 5.3; x64)',
      authorization: 'token [REDACTED]',
      'content-type': 'application/json; charset=utf-8'
    },
    body: '{"tag_name":"v1.0.0","name":"v1.0.0","body":"# 1.0.0 (2020-05-14)\\n\\n\\n### Features\\n\\n* implement initial version of vue-accessible-color-picker ([eba7d30](https://github.com/kleinfreund/vue-accessible-color-picker/commit/eba7d3055364c2bcbeaa1a0067df6521050d2d3b))\\n\\n\\n\\n","prerelease":false}',
    request: { agent: undefined, hook: [Function: bound bound register] }
  },
  pluginName: '@semantic-release/github'
}
RequestError [HttpError]: Cookies must be enabled to use GitHub.
    at /home/runner/work/vue-accessible-color-picker/vue-accessible-color-picker/node_modules/@octokit/request/dist-node/index.js:66:23
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  status: 403,
  headers: {
    'cache-control': 'no-cache',
    connection: 'close',
    'content-encoding': 'gzip',
    'content-security-policy': "default-src 'none'; base-uri 'self'; block-all-mixed-content; connect-src 'self' uploads.github.com www.githubstatus.com collector.githubapp.com api.github.com www.google-analytics.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com cdn.optimizely.com logx.optimizely.com/v1/events wss://live.github.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com; frame-ancestors 'none'; frame-src render.githubusercontent.com; img-src 'self' data: github.githubassets.com identicons.github.com collector.githubapp.com github-cloud.s3.amazonaws.com *.githubusercontent.com; manifest-src 'self'; media-src 'none'; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com",
    'content-type': 'text/plain; charset=utf-8',
    date: 'Thu, 14 May 2020 17:40:28 GMT',
    'expect-ct': 'max-age=2592000, report-uri="https://api.github.com/_private/browser/errors"',
    'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
    server: 'GitHub.com',
    'set-cookie': '_gh_sess=[REDACTED]; path=/; secure; HttpOnly',
    status: '403 Forbidden',
    'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
    'transfer-encoding': 'chunked',
    vary: 'X-PJAX, Accept-Encoding, Accept, X-Requested-With',
    'x-content-type-options': 'nosniff',
    'x-frame-options': 'deny',
    'x-github-request-id': '0400:0851:57C564:924595:5EBD828C',
    'x-xss-protection': '1; mode=block'
  },
  request: {
    method: 'POST',
    url: 'https://github.com/repos/kleinfreund/vue-accessible-color-picker/releases',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-rest.js/17.9.0 octokit-core.js/2.5.0 Node.js/12.16.3 (Linux 5.3; x64)',
      authorization: 'token [REDACTED]',
      'content-type': 'application/json; charset=utf-8'
    },
    body: '{"tag_name":"v1.0.0","name":"v1.0.0","body":"# 1.0.0 (2020-05-14)\\n\\n\\n### Features\\n\\n* implement initial version of vue-accessible-color-picker ([eba7d30](https://github.com/kleinfreund/vue-accessible-color-picker/commit/eba7d3055364c2bcbeaa1a0067df6521050d2d3b))\\n\\n\\n\\n","prerelease":false}',
    request: { agent: undefined, hook: [Function: bound bound register] }
  },
  pluginName: '@semantic-release/github'
}
##[error]Process completed with exit code 1.
</details>

## Expected behavior

<!-- Describe what the desired behavior would be. -->

For this error to not happened

or

for this error to actually tell me something useful (I know this is in particular not a problem with semantic-release).

## Environment

- **semantic-release** version: 17.0.7
- CI environment: GitHub actions
- Plugins used: *none*
- **semantic-release** configuration: https://github.com/kleinfreund/vue-accessible-color-picker/blob/main/release.config.js
- CI logs: https://github.com/kleinfreund/vue-accessible-color-picker/runs/675427677

--- a/README.md
+++ b/README.md
@@ -64,11 +64,11 @@ If you have actions that trigger on newly created releases, please use a generat
 
 ### Environment variables
 
-| Variable                       | Description                                               |
-| ------------------------------ | --------------------------------------------------------- |
-| `GH_TOKEN` or `GITHUB_TOKEN`   | **Required.** The token used to authenticate with GitHub. |
-| `GH_URL` or `GITHUB_URL`       | The GitHub Enterprise endpoint.                           |
-| `GH_PREFIX` or `GITHUB_PREFIX` | The GitHub Enterprise API prefix.                         |
+| Variable                                           | Description                                               |
+| -------------------------------------------------- | --------------------------------------------------------- |
+| `GH_TOKEN` or `GITHUB_TOKEN`                       | **Required.** The token used to authenticate with GitHub. |
+| `GITHUB_API_URL` or `GH_URL` or `GITHUB_URL`       | The GitHub Enterprise endpoint.                           |
+| `GH_PREFIX` or `GITHUB_PREFIX`                     | The GitHub Enterprise API prefix.                         |
 
 ### Options
 

--- a/lib/resolve-config.js
+++ b/lib/resolve-config.js
@@ -16,7 +16,7 @@ module.exports = (
   {env}
 ) => ({
   githubToken: env.GH_TOKEN || env.GITHUB_TOKEN,
-  githubUrl: githubUrl || env.GH_URL || env.GITHUB_URL,
+  githubUrl: githubUrl || env.GITHUB_API_URL || env.GH_URL || env.GITHUB_URL,
   githubApiPathPrefix: githubApiPathPrefix || env.GH_PREFIX || env.GITHUB_PREFIX || '',
   proxy: proxy || env.HTTP_PROXY,
   assets: assets ? castArray(assets) : assets,

--- a/test/helpers/mock-github.js
+++ b/test/helpers/mock-github.js
@@ -5,7 +5,7 @@ const nock = require('nock');
  *
  * @param {Object} [env={}] Environment variables.
  * @param {String} [githubToken=env.GH_TOKEN || env.GITHUB_TOKEN || 'GH_TOKEN'] The github token to return in the authentication response.
- * @param {String} [githubUrl=env.GH_URL || env.GITHUB_URL || 'https://api.github.com'] The url on which to intercept http requests.
+ * @param {String} [githubUrl=env.GITHUB_API_URL || env.GH_URL || env.GITHUB_URL || 'https://api.github.com'] The url on which to intercept http requests.
  * @param {String} [githubApiPathPrefix=env.GH_PREFIX || env.GITHUB_PREFIX || ''] The GitHub Enterprise API prefix.
  * @return {Object} A `nock` object ready to respond to a github authentication request.
  */
@@ -13,7 +13,7 @@ function authenticate(
   env = {},
   {
     githubToken = env.GH_TOKEN || env.GITHUB_TOKEN || 'GH_TOKEN',
-    githubUrl = env.GH_URL || env.GITHUB_URL || 'https://api.github.com',
+    githubUrl = env.GITHUB_API_URL || env.GH_URL || env.GITHUB_URL || 'https://api.github.com',
     githubApiPathPrefix = env.GH_PREFIX || env.GITHUB_PREFIX || '',
   } = {}
 ) {

--- a/test/publish.test.js
+++ b/test/publish.test.js
@@ -393,3 +393,45 @@ test.serial('Throw error without retries for 400 error', async (t) => {
   t.is(error.status, 400);
   t.true(github.isDone());
 });
+
+test.serial(
+  'Publish a release when env.GITHUB_URL is set to https://github.com (Default in GitHub Actions, #268)',
+  async (t) => {
+    const owner = 'test_user';
+    const repo = 'test_repo';
+    const env = {
+      GITHUB_TOKEN: 'github_token',
+      GITHUB_URL: 'https://github.com',
+      GITHUB_API_URL: 'https://api.github.com',
+    };
+    const pluginConfig = {};
+    const nextRelease = {gitTag: 'v1.0.0', name: 'v1.0.0', notes: 'Test release note body'};
+    const options = {repositoryUrl: `https://github.com/${owner}/${repo}.git`};
+    const releaseUrl = `https://github.com/${owner}/${repo}/releases/${nextRelease.version}`;
+    const releaseId = 1;
+    const uploadUri = `/api/uploads/repos/${owner}/${repo}/releases/${releaseId}/assets`;
+    const uploadUrl = `https://github.com${uploadUri}{?name,label}`;
+
+    const github = authenticate({...env, GITHUB_TOKEN: 'https://api.github.com'})
+      .post(`/repos/${owner}/${repo}/releases`, {
+        tag_name: nextRelease.gitTag,
+        name: nextRelease.name,
+        body: nextRelease.notes,
+        prerelease: false,
+      })
+      .reply(200, {upload_url: uploadUrl, html_url: releaseUrl});
+
+    const result = await publish(pluginConfig, {
+      cwd,
+      env,
+      options,
+      branch: {type: 'release', main: true},
+      nextRelease,
+      logger: t.context.logger,
+    });
+
+    t.is(result.url, releaseUrl);
+    t.deepEqual(t.context.log.args[0], ['Published GitHub release: %s', releaseUrl]);
+    t.true(github.isDone());
+  }
+);

--- a/test/publish.test.js
+++ b/test/publish.test.js
@@ -394,7 +394,7 @@ test.serial('Throw error without retries for 400 error', async (t) => {
   t.true(github.isDone());
 });
 
-test.serial(
+test.serial.only(
   'Publish a release when env.GITHUB_URL is set to https://github.com (Default in GitHub Actions, #268)',
   async (t) => {
     const owner = 'test_user';
@@ -412,7 +412,7 @@ test.serial(
     const uploadUri = `/api/uploads/repos/${owner}/${repo}/releases/${releaseId}/assets`;
     const uploadUrl = `https://github.com${uploadUri}{?name,label}`;
 
-    const github = authenticate({...env, GITHUB_TOKEN: 'https://api.github.com'})
+    const github = authenticate({...env, GITHUB_URL: 'https://api.github.com'})
       .post(`/repos/${owner}/${repo}/releases`, {
         tag_name: nextRelease.gitTag,
         name: nextRelease.name,

--- a/test/publish.test.js
+++ b/test/publish.test.js
@@ -394,7 +394,7 @@ test.serial('Throw error without retries for 400 error', async (t) => {
   t.true(github.isDone());
 });
 
-test.serial.only(
+test.serial(
   'Publish a release when env.GITHUB_URL is set to https://github.com (Default in GitHub Actions, #268)',
   async (t) => {
     const owner = 'test_user';
@@ -412,7 +412,7 @@ test.serial.only(
     const uploadUri = `/api/uploads/repos/${owner}/${repo}/releases/${releaseId}/assets`;
     const uploadUrl = `https://github.com${uploadUri}{?name,label}`;
 
-    const github = authenticate({...env, GITHUB_URL: 'https://api.github.com'})
+    const github = authenticate(env)
       .post(`/repos/${owner}/${repo}/releases`, {
         tag_name: nextRelease.gitTag,
         name: nextRelease.name,

